components/your-history/YourHistory.js

```diff
@@ -0,0 +1,98 @@
+import { makeStyles } from '@material-ui/styles';
+import { format, compareDesc, isSameMonth, isSameYear } from 'date-fns';
+import CalendarIcon from '@material-ui/icons/CalendarTodayRounded';
+import Grid from '@material-ui/core/Grid';
+import React, { useState, useEffect } from 'react';
+import Typography from '@material-ui/core/Typography';
+import ActivityCard from './ActivityCard';
+import YourHistoryPropTypes from './PropTypes';
+import ScaffoldContainer from '../ScaffoldContainer';
+
+const useStyles = makeStyles(theme => ({
+  root: {
+    padding: theme.spacing(5, 0),
+  },
+  pageHeader: {
+    marginTop: theme.spacing(5),
+    marginBottom: theme.spacing(5),
+    fontWeight: theme.typography.fontWeightMedium,
+  },
+  listItem: {
+    marginTop: theme.spacing(2),
+  },
+  sectionHeader: {
+    display: 'flex',
+    justifyContent: 'flex-start',
+    alignItems: 'center',
+    marginTop: theme.spacing(4),
+    marginBottom: theme.spacing(2),
+  },
+  calendarIcon: {
+    marginRight: theme.spacing(1),
+  },
+}));
+
+export default function YourHistory(props) {
+  const classes = useStyles();
+  const [visibleActivities, setVisibleActivities] = useState([]);
+  const [activityMonths, setActivityMonths] = useState([]);
+  const { activities } = props;
+
+  const isInMonthYear = (date, monthYear) =>
+    isSameMonth(date, monthYear) && isSameYear(date, monthYear);
+
+  useEffect(() => {
```

Comment:
I'm not sure we need `useEffect` here.

Something like this should work, without popping the escape hatch into the imperative world:

```diff
diff --git a/components/your-history/YourHistory.js b/components/your-history/YourHistory.js
index 5c3ca0cc..fa3a6292 100644
--- a/components/your-history/YourHistory.js
+++ b/components/your-history/YourHistory.js
@@ -34,14 +34,13 @@ const useStyles = makeStyles(theme => ({
 
 export default function YourHistory(props) {
   const classes = useStyles();
-  const [visibleActivities, setVisibleActivities] = useState([]);
-  const [activityMonths, setActivityMonths] = useState([]);
+  let visibleActivities = [];
+  let activityMonths = [];
   const { activities } = props;
 
   const isInMonthYear = (date, monthYear) =>
     isSameMonth(date, monthYear) && isSameYear(date, monthYear);
 
-  useEffect(() => {
   if (!activities.empty) {
     const dates = [];
     const temp = activities.map(a => {
@@ -52,10 +51,9 @@ export default function YourHistory(props) {
       }
       return activity;
     });
-      setVisibleActivities(temp);
-      setActivityMonths(dates);
+    visibleActivities = temp;
+    activityMonths = dates;
   }
-  }, [activities]); //eslint-disable-line
 
   return (
     <div className={classes.root}>
```

Reply:
good call